readthedocs/restapi/views/model_views.py

```diff
@@ -32,6 +33,25 @@
 log = logging.getLogger(__name__)
 
 
+class PlainTextRenderer(BaseRenderer):
+
+    """
+    Custom renderer for text/plain format.
+
+    charset is 'utf-8' by default.
+    """
+
+    media_type = 'text/plain'
+    format = 'txt'
+
+    def render(self, data, accepted_media_type=None, renderer_context=None):
+        kwargs = renderer_context.get('kwargs', {})
+        if kwargs.get('format', self.format) == self.format:
```

Comment:
I think this is not necessary.

The example doesn't handle this: http://www.django-rest-framework.org/api-guide/renderers/#example

Also, take a look at the renderes that comes with DRF: https://github.com/encode/django-rest-framework/blob/master/rest_framework/renderers.py

Reply:
I was getting a 500 error when requesting a file type distinct from txt here, so better to have a 404. I was not sure why the file type wasn't check automatically, probably something to do with the version of restframework or https://github.com/rtfd/readthedocs.org/pull/3585/files#r168669155.

Note the parameters are distinct from this version and from the example of the latest version.

Reply:
OK. I'd say that we need to research a little more here since I'm almost sure that this should be handled by DRF itself instead by us. Can you take a deeper look on this?

Try upgrading DRF to see if it's related to that or it's another thing. How other endpoint that only support JSON behaves when you pass `format=html`? Who handles that?

Those could be a good tests to have a better idea.

Reply:
I think I figure out, the renderer also is expected to give a representation of a `404` error (any status actually).